.swiftlint.yml

```diff
@@ -0,0 +1,83 @@
+opt_in_rules:
+  - anonymous_argument_in_multiline_closure
+  - array_init
+  - closure_end_indentation
+  - closure_spacing
+  - collection_alignment
+  - comma_inheritance
+  - contains_over_filter_count
+  - contains_over_filter_is_empty
+  - contains_over_first_not_nil
+  - contains_over_range_nil_comparison
+  - convenience_type
+  - direct_return
+  - discarded_notification_center_observer
+  - discouraged_none_name
+  - discouraged_object_literal
+  #- discouraged_optional_boolean # needs code change
+  - empty_collection_literal
+  - empty_count
+  - empty_string
+  - enum_case_associated_values_count
+  - expiring_todo
+  - explicit_init
+  # - explicit_type_interface # nice to have
+  - fallthrough
+  - fatal_error_message
+  - file_name_no_space
+  - first_where
+  - flatmap_over_map_reduce
+  # - force_unwrapping # nice to have
```

Comment:
Do we need some additional changes to introduce those marked as `nice to have`? Or do you want to enable them later?

Reply:
Oh, I should have commented on these rules. As you can see, I've added many opt-in rules. It's up to project maintainers to enable or disable them because these rules are quite opinionated. I've enabled rules, that don't contradict with current styles in codebase. I think, all commented out rules are also good to have, but I wanted you to decide. 

You can check description and examples [here](https://realm.github.io/SwiftLint/rule-directory.html).

Reply:
I went through the commented ones: 

- `no_grouping_extension` - Let's remove this one for now
- `no_magic_numbers` - this one looks great, I don't like magic numers 
- `nslocalizedstring_key` - Let's keep this one as I want to introduce translations one day 
- `implicitly_unwrapped_optional` - This would require changes in the `AppDelegate` and `Menu` file - let's keep it commented and we can change it in another PR
- `force_unwrapping` - same as above
- `explicit_type_interface` - I dont like this one
- `discouraged_optional_boolean` - This requires changes so let's skip it for now

Reply:
`no_magic_numbers` adds quite a lot of warnings (~60). But it actually shows some code smell in views. 🙃 I would say, let's keep it disabled and create another task to address.

Reply:
Okay! Let's keep it disabled